scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+-----------------------------------
+-- PET: Avatar
+-----------------------------------
+require('scripts/globals/summon')
+-----------------------------------
+
+xi = xi or {}
+xi.pets = xi.pets or {}
+xi.pets.avatar = {}
+
+local buffModeVar          = 'AVATAR_BUFF_MODE_OFF'
+local lastCastTimeVar      = 'AVATAR_LAST_CASTINGTIME'
+local lastCastTimeStampVar = 'AVATAR_LAST_CAST_TIMESTAMP'
+local playerListenerVar    = 'SMN_SPIRIT_CAST_DELAY'
+
+-- will determine, in seconds, the spirit's casting cooldown
+-- Called every tick for player to adjust magic casting delay in real-time
+local setMagicCastCooldown = function(pet)
+    local castingCooldown = 45
+    local master = pet:getMaster()
+
+    -- we already know master is a PC, this is just in case the pet gets disjointed somehow
+    if master then
+        castingCooldown = castingCooldown - math.floor(xi.summon.getSummoningSkillOverCap(pet) / 3)
+
+        if master:hasStatusEffect(xi.effect.ASTRAL_FLOW) then
+            castingCooldown = castingCooldown - 5
+        end
+
+        local masterLegs = master:getEquipID(xi.slot.LEGS)
+        if
+            masterLegs and
+            (masterLegs == xi.item.SUMMONERS_SPATS or
```

Comment:
this should be a mod added to these items

Reply:
Was on the fence about that since it's the only item in the game with the effect (along with its +X)

I'll add the mod